pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -673,7 +730,8 @@ func (nc *Controller) doEvictionPass() {
 				klog.Warningf("Failed to get Node %v from the nodeLister: %v", value.Value, err)
 			}
 			nodeUID, _ := value.UID.(string)
-			remaining, err := nodeutil.DeletePods(nc.kubeClient, nc.recorder, value.Value, nodeUID, nc.daemonSetStore)
+			pods := nc.podsToEvict.getAndRemove(value.Value)
+			remaining, err := nodeutil.DeletePods(nc.kubeClient, pods, nc.recorder, value.Value, nodeUID, nc.daemonSetStore)
```

Comment:
This may have significant impact on how we rate limit.
Imagine that we're slowly delivering updates of pods - then for each of those we may potentially need to run a separate eviction pass (in practive probably not super probable, but still).
I would like to see the explanation why we believe it's fine.

Conceptually, what would be more correct is: if a node has already been processed by doEvictionPass, newly added pod is immediately evicted to (without really waiting for the next eviction pass).

Reply:
@krzysied - has this been addressed?

Reply:
I changed `podToEvictionMap` to `nodeEvictionMap`. It contains list of pods to be evicted + node eviction status.
If node eviction status == evicted, pods are deleted immediately.
nodeEvictionMap is protected by evictorLock.